src/Pelagos/Util/DOIutil.php

```diff
@@ -375,4 +334,51 @@ function ($matches) {
             $input
         );
     }
+
+    /**
+     * This function generates a new DOI string that is not already in use per datacite.
+     *
+     * @return string A unique, unused DOI string.
+     */
+    public function generateDoi()
+    {
+        $encoder = new Base32CrockfordEncoder();
+        // 1099511627775 encodes to the longest 8 character Crockford 32 string.
+        $max = 1099511627775;
+        // Start at 1 (0 is problematic as library does not produce checksum for 0.)
+        $random = mt_rand(1, $max);
+        // Add shoulder and remove the checksum character.
+        $doi = $this->doishoulder . '/' . substr($encoder->encode($random), 0, -1);
```

Comment:
The shoulder on prod is 10.7266/n7, so that '/' can be removed

Reply:
I didn't realize N7 was part of our shoulder! I'll have to shorten the DOI generator by 2 characters.

Reply:
If I understand correctly, we own the entire doi prefix now.

Reply:
We always owned the entire prefix, the shoulder was just a EZID thing, carried over to EZAPI for compatibility.  Creating a DOI we can ditch the N7 if we want  See https://ezid.cdlib.org/learn/id_basics for more details

Reply:
/N7 was not part of our prefix, and we really intermixed the wording on shoulder vs prefix in our prior code. I checked with the PO and we don't need to maintain the /N7 thing as that was specific to ezid's. All we need is 8 characters after the assigned prefix.